src/Expander.hs

```diff
@@ -988,12 +992,12 @@ expandOneForm prob stx
           saveExprType dest t
         EConstructor ctor -> do
           ConstructorInfo args dt <- constructorInfo ctor
-          DatatypeInfo arity _ <- datatypeInfo dt
+          DatatypeInfo argKinds _ <- datatypeInfo dt
           case prob of
             ExprDest t dest -> do
-              argTys <- makeTypeMetas arity
+              argTys <- traverse makeTypeMeta argKinds
```

Comment:
I am passing the kind to `makeTypeMeta`, but it currently ignores it; eventually, we'll want to associate a kind to each type variable (and perhaps even to infer their kinds).

Reply:
Yeah, I suspect we'll actually want to associate a fresh kind variable with each type variable once we make some more progress.